docs/tutorial/index.rst

```diff
@@ -42,3 +42,8 @@ Extended tutorial
 
 Once you have completed the core tutorial, the extended optional tutorial
 sections can be followed in any order - they don’t depend on each other.
+
+.. toctree::
+   :maxdepth: 1
+
+  writing-tests/test-case
```

Comment:
Suggestion:
I wonder if it'd be helpful to split the sections into new pages and list them in sequence, so that the document is easier to navigate and maintain. (like in `docs/reference/units/`)
Also, some of the sections are useful in other scenarios as well (e.g. defining resources, unit testing, complexity.), so it'd be great to have them appear in the table of contents panel for discoverability.

Reply:
There's also an option to show the ordered number before the title without changing the files.
```
.. toctree::
   :maxdepth: 1
   :numbered:
```

for example:
![image](https://github.com/user-attachments/assets/7e2e542d-e483-4149-8b50-c2178a669b6e)

Reply:
I think this tutorial is better takes as a full, 1 session lesson on checkbox tests, so I would rather keep it as a single big document